### DISCUSSION TITLE
Request for UriHelper interface (i.e. IUriHelper)?

### DISCUSSION THREAD

**Discussion by aer0ace** [atteneder/glTFast#448] [Ideas]
For a project I'm working on, the URI specified to load glTF models from a server is more complex than just a path to the model. As such, I've experimented with modifying UriHelper to suit our needs, but since it's part of the glTFast source, I can't override default behavior with some sort of DI solution. It'd be great if we can have an IUriHelper, with the existing implementation as the DefaultUriHelper, so that we can implement our own solution and inject it into GltfImport from the consuming code.

**Comment by atteneder** [comment]
Hi @aer0ace,

Sounds doable, but first I'd like to understand if this is really project specific or in fact something that could be implemented within glTFast.

Which of `UriHelper`'s methods would you like to override? Can you give an (if required, obfuscated) example of the URIs you're having struggle with?

Thanks

**Reply by aer0ace** [reply]
Our project requires a URI that specifies URL parameters. For example, http://localhost:8080/service/?file=specialPathtoFile&type=gltf.
I know it may be "unconventional", but the hard-coded UriHelper is what's stopping us from doing this.
My local solution was to extract an interface from UriHelper, into IUriHelper. For our solution we only needed to override GetBaseUri() and GetUriString(), and that was enough to support custom URIs, specifically to support URL parameters.
